src/utils/uploader.js

```diff
@@ -0,0 +1,425 @@
+        const fileSize = size;
+        const hexName = utils.stringToHex(name);
+        const fileContract = new ethers.Contract(this.#contractAddress, FlatDirectoryAbi, this.#wallet);
+        const fileMod = await fileContract.getStorageMode(hexName);
+        if (fileMod !== BigInt(VERSION_CALL_DATA) && fileMod !== 0n) {
+            throw new Error(`FlatDirectory: This file does not support calldata upload!`);
+        }
+
+        let chunkDataSize = fileSize;
+        let chunkLength = 1;
+        if (GALILEO_CHAIN_ID === this.#chainId) {
+            if (fileSize > 475 * 1024) {
+                // Data need to be sliced if file > 475K
+                chunkDataSize = 475 * 1024;
+                chunkLength = Math.ceil(fileSize / (475 * 1024));
+            }
+        } else {
+            if (fileSize > 24 * 1024 - 326) {
+                // Data need to be sliced if file > 24K
+                chunkDataSize = 24 * 1024 - 326;
+                chunkLength = Math.ceil(fileSize / (24 * 1024 - 326));
+            }
+        }
+
+        let totalStorageCost = 0n;
+        let totalGasCost = 0n;
+        let gasLimit = 0;
+        const oldChunkLength = await fileContract.countChunks(hexName);
+        for (let i = 0; i < chunkLength; i++) {
+            const chunk = getFileChunk(path, fileSize, i * chunkDataSize, (i + 1) * chunkDataSize);
+
+            // check is change
+            if (oldChunkLength !== 0 && i < oldChunkLength) {
+                const localHash = '0x' + sha3(chunk);
+                const hash = await fileContract.getChunkHash(hexName, i);
+                if (localHash === hash) {
+                    continue;
+                }
+            }
+
+            // get cost
+            let cost = 0n;
+            if (GALILEO_CHAIN_ID === this.#chainId && chunk.length > (24 * 1024 - 326)) {
+                // Galileo need stake
+                cost = Math.floor((chunk.length + 326) / 1024 / 24);
+                cost = ethers.parseEther(cost.toString());
+            }
+            if (i === chunkLength - 1 || gasLimit === 0) {
+                const hexData = '0x' + chunk.toString('hex');
+                gasLimit = await fileContract.writeChunk.estimateGas(hexName, 0, hexData, {
+                    value: cost
+                });
+            }
+            totalStorageCost += cost;
+            totalGasCost += (gasFeeData.maxFeePerGas + gasFeeData.maxPriorityFeePerGas) * BigInt(100 + gasPriceIncreasePercentage) / BigInt(100) * gasLimit;
+        }
+
+        return {
+            totalStorageCost, totalGasCost
+        }
+    }
+
+    // upload
+    async upload(path, gasPriceIncreasePercentage = 0) {
+        const syncPoolSize = this.#uploadType === VERSION_BLOB ? 3 : 5;
+        const results = [];
+        return new Promise((resolve, reject) => {
+            from(recursiveFiles(path, ''))
+                .pipe(mergeMap(info => this.#upload(info, gasPriceIncreasePercentage), syncPoolSize))
```

Comment:
'this.#upload(info, gasPriceIncreasePercentage)` is confusing because the second parameter is `syncPoolSize`

Reply:
Also what does syncPoolSize mean?